src/MaxText/pyconfig.py

```diff
@@ -1119,6 +1122,12 @@ def validate_ragged_dot(raw_keys):
       max_logging.log(f"JAX config {config_flag} not found, possibly due to old JAX version.")
 
 
+def validate_optimizer_sharding_over_data(raw_keys):
+  zero1_supported_opt_types = ("adamw", "adam_pax")
+  if raw_keys["opt_type"] not in zero1_supported_opt_types:
+    raise ValueError(f"Optimizer type {raw_keys["opt_type"]} is not supported for optimizer sharding.")
```

Comment:
Nit: You could add an extra sentence saying `Please use an optimizer from this list: {zero1_supported_opt_types}`